dandi/cli/cmd_download.py

```diff
@@ -67,6 +67,11 @@ def get_metavar(self, param):
     type=click.Choice(["pyout", "debug"]),
     default="pyout",
 )
+@click.option(
+    "--glob",
+    is_flag=True,
+    help="Interpret asset paths in URLs as path glob patterns",
+)
```

Comment:
@yarikoptic Is the non-`glob` choice for `--path-type` supposed to be `exact` or `explicit`?  Your original comment in #1255 says "`exact`", and your comment here says both.  (I think `exact` is the better name.)

> automatically switch from "exact" to "glob" if URL explicitly asks for `glob`

The way the PR currently works, a `/?glob=` URL will always be expanded in "glob" mode even if `--glob`/`--path-type=glob` is not given.  Moreover, if `--glob` is not given, the presence of a `/?glob=` URL will not affect the interpretation of `/?location=` etc. URLs also given on the command line.  Are you asking here for the first behavior (which is already implemented) or a change to the second behavior (which seems like a bad idea)?

Reply:
yeah, I originally used "exact" but then apparently incompletely switched to "explicit" since I felt it to be a better choice. But if you like "exact" -- fine with me.

> The way the PR currently works, a `/?glob=` URL will always be expanded in "glob" mode even if `--glob`/`--path-type=glob` is not given.

and that is good -- that is why I liked "explicit" as URL alone explicitly defines which method to use.

> Moreover, if `--glob` is not given, the presence of a `/?glob=` URL will not affect the interpretation of `/?location=` etc.

If I got it right (that one URL doesn't effect interpretation of another) -- then it is great!  I was asking for the first (was not thinking about multiple URLs) -- I just meant that what you said. So all good.